src/models/FundCampaign.ts

```diff
@@ -0,0 +1,85 @@
+import type { Types, Model, PopulatedDoc, Document } from "mongoose";
+import { Schema, model, models } from "mongoose";
+import type { InterfaceUser } from "./User";
+import type { InterfaceFund } from "./Fund";
+import type { InterfaceCampaignPledge } from "./CampaignPledge";
+
+/**
+ * This is an interface representing a document for a Fund Campaign in the database(MongoDB).
+ */
+
+export interface InterfaceFundCampaign {
+  _id: Types.ObjectId | string;
+  name: string;
+  creatorId: PopulatedDoc<InterfaceUser & Document>;
+  goalAmount: number;
+  currency: string;
+  startDate: Date;
+  endDate: Date;
+  createdAt: Date;
+  updatedAt: Date;
+  parentFund: PopulatedDoc<InterfaceFund & Document>;
+  pledgeId: PopulatedDoc<InterfaceCampaignPledge & Document>;
```

Comment:
Many fields are repeated between the models `Fund`, `FundCampaign` and `CampaignPlegde`. The relationship should be modelled through parent and children, and only the id of the parent should be stored on the children to reduce data redundancy.